parlai/core/params.py

```diff
@@ -1285,12 +1303,38 @@ def _handle_custom_options(self, kwargs):
             kwargs['type'] = 'bool'
         return kwargs, action_attr
 
+    def _handle_single_dash_addarg(self, args):
+        """
+        Fixup argparse for parlai-style short args.
+
+        In python 3.8, argparsing was changed such that short arguments are not
+        required to have spaces after them. This causes our many short args to
+        be misinterpetted by the parser. For example `-emb` gets parsed as
+        `-e mb`.
+
+        Here we rewrite them into long args to get around the nonsense.
+        """
+        if _sys.version_info < (3, 8, 0):
+            # older python works fine
+            return args
+
+        out_long = []
+        out_short = []
```

Comment:
nit: why can't  they both use the same list? You returned their aggregated version only anyway.

Reply:
bc i'm being cute and swapping the ordering

Reply:
(we tend to define options like `add_argument('-hs', '--hiddensize')` but we need it to `add_argument('--hiddensize', `--hs`) or otherwise it will try to store the value to `opt['hs']`

Reply:
Oh, I see. That makes sense.